sentry/src/main/java/io/sentry/Baggage.java

```diff
@@ -440,38 +446,34 @@ public void setReplayId(final @Nullable String replayId) {
     set(DSCKeys.REPLAY_ID, replayId);
   }
 
-  @ApiStatus.Internal
-  public void set(final @NotNull String key, final @Nullable String value) {
-    set(key, value, false);
-  }
-
   /**
-   * Sets / updates a value
+   * Sets / updates a value, but only if the baggage is still mutable.
    *
    * @param key key
    * @param value value to set
-   * @param force ignores mutability of this baggage and sets the value anyways
    */
-  private void set(final @NotNull String key, final @Nullable String value, final boolean force) {
-    if (mutable || force) {
+  @ApiStatus.Internal
+  public void set(final @NotNull String key, final @Nullable String value) {
+    if (mutable) {
       this.keyValues.put(key, value);
     }
   }
 
   @ApiStatus.Internal
   public @NotNull Map<String, Object> getUnknown() {
     final @NotNull Map<String, Object> unknown = new ConcurrentHashMap<>();
-    for (Map.Entry<String, String> keyValue : this.keyValues.entrySet()) {
-      final @NotNull String key = keyValue.getKey();
-      final @Nullable String value = keyValue.getValue();
-      if (!DSCKeys.ALL.contains(key)) {
-        if (value != null) {
-          final @NotNull String unknownKey = key.replaceFirst(SENTRY_BAGGAGE_PREFIX, "");
-          unknown.put(unknownKey, value);
+    synchronized (keyValues) {
```

Comment:
We should use `AutoClosableReentrantLock` over `synchronized` as there was problems with virtual threads and synchronized before.